OpenRA.Mods.Common/Traits/GainsExperience.cs

```diff
@@ -60,16 +63,18 @@ public GainsExperience(ActorInitializer init, GainsExperienceInfo info)
 			this.info = info;
 
 			MaxLevel = info.Conditions.Count;
-			var cost = self.Info.TraitInfo<ValuedInfo>().Cost;
-			foreach (var kv in info.Conditions)
-				nextLevel.Add(Pair.New(kv.Key * cost, kv.Value));
 
 			if (init.Contains<ExperienceInit>())
 				initialExperience = init.Get<ExperienceInit, int>();
 		}
 
 		void INotifyCreated.Created(Actor self)
 		{
+			var valued = self.Info.TraitInfoOrDefault<ValuedInfo>();
+			var requiredExperience = info.RequiredExperience == -1 ? (valued != null ? valued.Cost : 0) : info.RequiredExperience;
```

Comment:
Would a negative `RequiredExperience` make sense at all? If not, please change this to `info.RequiredExperience < 0`.

Reply:
@IceReaper: would you mind changing this too?